### PR TITLE
feat(Button): added download prop

### PR DESCRIPTION
**Pull request by nicmell**
Added download prop to indicate the href attribute refers to a file to be downloaded rather than displayed in the browser

### CODE REVIEW

File: src/components/Button/Button.props.ts [hunk 1]
```diff
@@ -129,4 +129,10 @@ export type ButtonProps = {
    * - "link": button link without padding and margin;
    */
   type?: Type,
+
+  /**
+   * Indicates that the linked resource is intended to be downloaded rather than displayed in the browser.
+   * If is a string, the values specifies the default file name for use in labeling the resource in a local file system.
```
Comment (malta895) [inline src/components/Button/Button.props.ts:135]:
```suggestion
   * If it is a string, the values specifies the default file name for use in labeling the resource in a local file system.
```

**Comment by fredmaggiowski** [comment]
How does this interact with the onClick prop? can you add at least on story to showcase the feature and how to use it?

**Comment by nicmell** [comment]
@fredmaggiowski I've gust added a story for the download prop.
The behaviour when both the onClick and download prop are set is:

the onClick callback is run, and the file referenced in the href is downloaded, unless the preventDefault() method on clickEvent passed as parameter to the callback is invoked

File: src/components/Button/Button.props.ts (hunk 1 above)
Comment (nicmell) [reply src/components/Button/Button.props.ts]: fixed in [[d1e589e](https://github.com/mia-platform/design-system/pull/621/commits/d1e589e2a7c23e659ef08dbecb80a9092dbc8701)](https://github.com/mia-platform/design-system/pull/621/commits/d1e589e2a7c23e659ef08dbecb80a9092dbc8701)

**Review by fredmaggiowski — APPROVED** [review]
LGTM

**Pull request merged**